src/Marten.Testing/DevelopmentModeRegistry.cs

```diff
@@ -10,7 +10,10 @@ public class DevelopmentModeRegistry : Registry
         public DevelopmentModeRegistry()
         {
             For<IConnectionFactory>().Use<ConnectionSource>();
-            ForSingletonOf<IDocumentSchema>().Use<Marten.Schema.DocumentSchema>();
+            ForSingletonOf<DocumentSchema>()
+                .Use<DocumentSchema>()
+                .OnCreation("", (ctx, schema) => schema.UsePostgres9_4Fallback = true);
+            Forward<DocumentSchema, IDocumentSchema>();
```

Comment:
`DocumentSchemaTests` works directly with the concrete `DocumentSchema` so the fallback mechanism has to be configured for the concrete type as well.